### PR TITLE
Check that the ABIs and bytecode from tasks matches what's in the build-info

### PR DESCRIPTION
**Pull request by TomAFrench**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -224,6 +224,28 @@ jobs:
       - name: Test
         run: yarn workspace @balancer-labs/v2-governance-scripts test
 
+  check-artifacts:
```
Comment (nventuro) [inline .github/workflows/ci.yml:227]: Maybe rename these to 'check-deployment-artifacts' and 'check-deployment-addresses'

File: pkg/deployments/tasks/20210418-vault/abi/BalancerHelpers.json [hunk 2]
```diff
@@ -68,7 +68,7 @@
         "type": "uint256[]"
       }
     ],
-    "stateMutability": "view",
+    "stateMutability": "nonpayable",
```
Comment (nventuro) [inline pkg/deployments/tasks/20210418-vault/abi/BalancerHelpers.json:71]: Ah unfortunate. We have the `overrideQueryFunctions` helper change this, but not on the build info itself.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (TomAFrench) [reply .github/workflows/ci.yml]: Done, I don't really want to change the hardhat tasks though as it'll be unwieldy on the CLI.

Comment (nventuro) [reply .github/workflows/ci.yml]: That's fine - the deployments context is already there since those belong to that package

**Pull request merged**